### PR TITLE
Adding user availability through Google Calendar API

### PR DESCRIPTION
**Pull request by fangyu3**
Storing Google tokens for Oauth

Update dependencies

Adding getting availability from Google API feature

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,4 +32,6 @@ instance/
 
 #TODO add .env to your gitignore
 .env
-/server/.env
+./server/.env
+./client/.env
+server/utils/auth/client_secrets.json
```
Comment (fangyu3) [inline .gitignore:37]: @bugatti12345 I will send the client_secret in slack, just put in under the same path

You'll also need to add below variables to your server.env. You can find the value in the client_secret.json
GOOGLE_CLIENT_ID 
GOOGLE_CLIENT_SECRET
GOOGLE_TOKEN_URL

**Comment by fangyu3** [comment]
This PR takes account of both #2 and #35

File: client/src/pages/SignUp.js [hunk 2]
```diff
@@ -67,13 +67,19 @@ function Signup() {
     const history = useHistory();
     const user = useContext(UserContext);
     const responseGoogle = (response) => {
+
+        // TODO: REMOVE LATER
+        for (const property in response) {
+            console.log(`${property}: ${response[property]}`);
+          }
+
```
Comment (johnsciutto) [inline client/src/pages/SignUp.js]: Remember to remove this.

File: client/src/pages/SignUp.js [hunk 3]
```diff
@@ -67,13 +67,19 @@ function Signup() {
     const history = useHistory();
     const user = useContext(UserContext);
     const responseGoogle = (response) => {
+
+        // TODO: REMOVE LATER
+        for (const property in response) {
```
Comment (sina-jamshidi) [inline client/src/pages/SignUp.js]: another reminder to remove this lol

File: server/api/google_signup_handler.py [hunk 4]
```diff
@@ -10,19 +10,33 @@
...
 @google_signup_handler.route('/googlesignup', methods=['POST'])
 def googlesignup():
-    token_id = json.loads(request.get_data())['tokenId']
-    user_info = oauth(token_id)
+    code = json.loads(request.get_data())['code']
+    google_client = GoogleClient(google_auth_code = code)
+    user_info = google_client.get_user_info()
 
-    if user_info['status']:
+    # print(user_info)
```
Comment (sina-jamshidi) [inline server/api/google_signup_handler.py]: remove comment

File: server/api/google_signup_handler.py [hunk 5]
```diff
@@ -10,19 +10,33 @@
...
-            user_db = Users(name=name, email=email, username=None, password_hash=None, google_id=userid, url='')
+            user_db = Users(name=name, 
+                            email=email, 
+                            username=None, 
+                            password_hash=None, 
+                            google_id=userid, 
+                            url='',
+                            access_token=user_info.get('access_token'),
+                            refresh_token=user_info.get('refresh_token')
+                            )
```
Comment (sina-jamshidi) [inline server/api/google_signup_handler.py:30]: 👍

File: server/utils/auth/google_client.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                    # "groupExpansionMax": integer,
+                    # "calendarExpansionMax": integer,
+                    "items": calendar_ids
+                }
+            
+            calendar_info = calendar_info_service.freebusy().query(body=time).execute().get("calendars",{})
+
+            busy_time = [time for outer in [v.get('busy') for (k,v) in calendar_info.items() if len(v.get('busy')) > 0] for time in outer] 
+            
+            return busy_time
```
Comment (sina-jamshidi) [inline server/utils/auth/google_client.py:85]: 👍

**Review by sina-jamshidi — APPROVED** [review]

**Pull request merged**